File: javascript/swapTwoNumbers.js
let a, b;

function swapTwoNumbers(a, b) {
  let temp;
  temp = a;
  a = b;
  b = temp;
}

function getValidNumber(promptMessage) {
  let num;
  do {
    num = prompt(promptMessage);
  } while (isNaN(num) || num.trim() === "");
  return parseFloat(num);
}

a = getValidNumber("Enter the first number: ");
b = getValidNumber("Enter the second number: ");

console.log("Before swapping: a= " + a + " b= " + b);

swapTwoNumbers(a, b);
console.log("After swapping: a= " + a + " b= " + b);

File: cSharp/swapTwoNumbers.cs
namespace swap
{
    public class Program
    {
        static void Main(string[] args)
        {
            int a, b;
            while (true)
            {
                Console.WriteLine("Enter first number");
                var success = int.TryParse(Console.ReadLine(), out a);
                if (success)
                {
                    break;
                }
                Console.WriteLine("Please enter a valid number");


            }
            while (true)
            {
                Console.WriteLine("Enter second number");
                var success = int.TryParse(Console.ReadLine(), out b);
                if (success)
                {
                    break;
                }
                Console.WriteLine("Please enter a valid number");

            }
            Console.WriteLine("Before swap a = {0} and b = {1}", a, b);

            Swap( ref a,ref b);
            Console.WriteLine("After swap a = {0} and b = {1}", a, b);


        }
        static void Swap( ref int a, ref int b)
        {
            int temp = a;
            a = b;
            b = temp;
        }
    }
}
